fontbe/src/features/kern.rs

```diff
@@ -359,31 +390,843 @@ impl Work<Context, AnyWorkId, Error> for KerningGatherWork {
+            )
+        }
+    }
+
+    #[test]
+    fn split_latin_and_cyrillic() {
+        const A_CY: char = 'а';
+        const BE_CY: char = 'б';
+        let charmap: MockCharMap = ['a', 'b', A_CY, BE_CY].into_iter().collect();
+        let known_scripts = scripts_for_chars(&charmap);
+        let pairs = [('a', 'b', 5i16), ('a', 'a', 7), (A_CY, BE_CY, 12)]
+            .into_iter()
+            .map(|(a, b, val)| charmap.make_rule(a, b, val))
+            .collect::<Vec<_>>();
+        let ctx =
+            KernSplitContext::new(&charmap, &known_scripts, None, Default::default()).unwrap();
+
+        let pairs_ref = pairs.iter().collect::<Vec<_>>();
+
+        let result = ctx.make_lookups(&pairs_ref);
+        assert_eq!(result.len(), 2);
+
+        let cyrillic: BTreeSet<_> = [UnicodeShortName::from_str("Cyrl").unwrap()]
+            .into_iter()
+            .collect();
+
+        let latn: BTreeSet<_> = [LATN].into_iter().collect();
+
+        let cyr_rules = result.get(&cyrillic).unwrap();
+        assert_eq!(
+            cyr_rules
+                .iter()
+                .flat_map(|x| x.subtables.iter().map(|sub| sub.len()))
+                .sum::<usize>(),
+            1
+        );
+
+        let latn_rules = result.get(&latn).unwrap();
+        assert_eq!(
+            latn_rules
+                .iter()
+                .flat_map(|x| x.subtables.iter().map(|sub| sub.len()))
+                .sum::<usize>(),
+            2
+        );
+    }
+
+    #[test]
+    fn mark_to_base_kern() {
+        const ACUTE_COMB: char = '\u{0301}';
+        let charmap: MockCharMap = ['A', 'B', 'C', ACUTE_COMB].into_iter().collect();
+        let known_scripts = scripts_for_chars(&charmap);
+        let mark_glyphs = [(charmap.get(ACUTE_COMB), MarkSpacing::NonSpacing)]
+            .into_iter()
+            .collect();
+        let pairs = [('A', ACUTE_COMB, -55), ('B', 'C', -30), ('A', 'C', -22)]
+            .into_iter()
+            .map(|(a, b, val)| charmap.make_rule(a, b, val))
+            .collect::<Vec<_>>();
+
+        let ctx = KernSplitContext::new(&charmap, &known_scripts, None, mark_glyphs).unwrap();
+
+        let pairs_ref = pairs.iter().collect::<Vec<_>>();
+        let result = ctx.make_lookups(&pairs_ref);
+
+        let latn: BTreeSet<_> = [LATN].into_iter().collect();
+        assert_eq!(result.len(), 1);
+        let lookups = result.get(&latn).unwrap();
+        assert_eq!(lookups.len(), 2);
+
+        let mut ignore_marks = LookupFlag::empty();
+        ignore_marks.set_ignore_marks(true);
+
+        let bases = &lookups[0];
+        assert_eq!(bases.flags, ignore_marks);
+        assert_eq!(
+            bases.subtables.iter().map(|sub| sub.len()).sum::<usize>(),
+            2
+        );
+
+        let marks = &lookups[1];
+        assert_eq!(marks.flags, LookupFlag::empty());
+        assert_eq!(
+            marks.subtables.iter().map(|sub| sub.len()).sum::<usize>(),
+            1
+        );
```

Comment:
This looks like it could be one assert and thus give more context when it fails. 

```rust
assert_eq!(
  vec![(ignore_marks, 2), (LookupFlag::empty())], lookups.iter()...etc);
```

mark_to_base_only below seems similar, so much so you could hoist out a helper to check you got one lookup with these flags and sizes.